### ISSUE TITLE
 treat empty strings as null values and vice versa

### ISSUE THREAD

**Issue by britzke** [Fabrik/fabrik#1361]
I just posted an idea to possibly treat empty strings as null values and vice versa at http://fabrik.uservoice.com/forums/15452-general/suggestions/6914156-empty-string-values-should-be-treated-as-null-and.

The simplest way to handle empty strings as null values is to set a global flag which can be configured by the user which advices fabrik to globally treat empty string values as null and vice versa.
A much better handling could be done on element basis where every element gets a property 'empty string is null' which could be automatically populated by the database 'nullable' property of that element. This is the preferred way to implement it.
I think you want to vote for it, too!

**Comment by olafzieger** [comment]
For me, this problem also occurs.

**Comment by yuriyua** [comment]
What is the reason of this method? We need NULL empty value in foreign keys. 

https://github.com/Fabrik/fabrik/blob/master/plugins/fabrik_element/cascadingdropdown/cascadingdropdown.php#L1269-L1287

```php
	/**
	 * Use in list model storeRow() to determine if data should be stored.
	 * Currently only supported for db join elements whose values are default values
	 * avoids casing '' into 0 for int fields
	 *
	 * Extended this from dbjoin element as an empty string should be possible in cdd, if no options selected.
	 * Otherwise previously selected values are kept
	 *
	 * @param   array  $data  Data being inserted
	 * @param   mixed  $val   Element value to insert into table
	 *
	 * @since   3.0.7
	 *
	 * @return boolean
	 */
	public function dataIsNull($data, $val)
	{
		return false;
	}
```

**Comment by Sophist-UK** [comment]
Exactly - that is why the corresponding method in the parent databasejoin object is overridden by this function which returns false (saying that the data is "not null" even if it  actually is.

https://github.com/Fabrik/fabrik/blob/master/plugins/fabrik_element/databasejoin/databasejoin.php#L3901-L3924

**Comment by britzke** [comment]
Yes! but when you update a record, all null valued attributes get the corresponding 0-value (may be 00-000-0000, if it is a date). So my issue is not specified exactly (sorry). It should rather be „null“ attributes should not change to the corresponding 0-value when updating a record“.

**Comment by Sophist-UK** [comment]
My point is that this is not a consequence of the code you highlighted.

**Comment by britzke** [comment]
from which code it come from? Is this a wanted feature? How to suppress the change of null-values when saving records?

**Comment by yuriyua** [comment]
> Exactly - that is why the corresponding method in the parent databasejoin object is overridden by this function which returns false (saying that the data is "not null" even if it actually is.

> https://github.com/Fabrik/fabrik/blob/master/plugins/fabrik_element/databasejoin/databasejoin.php#L3901-L3924

Yes, but I need NULL value on empty field in the element cascadingdropdown. I need execute this method from the parent object.

**Comment by britzke** [comment]
We need NULL values not only for database join elements but for all elements that can have NULL values in the database. It seems that @olafzieger needs this, too.

**Comment by jaanusnurmoja** [comment]
Is there any function that stores data in db table? 
Just 
return;
stores NULL in database field.
Like does the function _indStoreDBFormat($val) in birthday element.
